src/GTP.cpp

```diff
@@ -882,10 +893,64 @@ bool GTP::execute(GameState & game, std::string xinput) {
         } else {
             gtp_fail_printf(id, "syntax not understood");
         }
+        return true;
+    } else if (command.find("set_option") == 0) {
+        std::istringstream cmdstream(command);
+        std::string tmp, option;
+
+        cmdstream >> tmp >> option;
+
+        if (!cmdstream.fail()) {
+            if (option.find("max_tree_size") == 0) {
+                int max_tree_size;
+                cmdstream >> max_tree_size;
+
+                if (!cmdstream.fail()) {
+                    cfg_max_tree_size = max_tree_size * 1048576LL; // * MB
+                    // 0 may be specified to mean "default"
+                    if (cfg_max_tree_size == 0) {
+                        cfg_max_tree_size = UCTSearch::MAX_TREE_SIZE;
+                    }
+                    gtp_printf(id, "Set max tree size to %d MB (%lld bytes)\n", cfg_max_tree_size / 1048576LL, cfg_max_tree_size);
+                    return true;
+                } else {
+                    gtp_fail_printf(id, "incorrect value for max_tree size");
+                    return true;
+                }
 
+            } else if (option.find("max_cache_size") == 0) {
+                int max_cache_size_mb;
+                cmdstream >> max_cache_size_mb;
+
+                if (!cmdstream.fail()) {
+                    size_t max_cache_size = max_cache_size_mb * 1048576LL; // * MB
+                    cfg_max_cache_count = max_cache_size / NNCache::ENTRY_SIZE;
+
+                    // 0 may be specified to mean "default"
+                    if (cfg_max_cache_count == 0) {
+                        cfg_max_cache_count = NNCache::MAX_CACHE_COUNT;
+                    }
+                    // don't allow less than default minimum
+                    if (cfg_max_cache_count < NNCache::MIN_CACHE_COUNT) {
+                        cfg_max_cache_count = NNCache::MIN_CACHE_COUNT;
+                        gtp_printf(id, "Too small value. Setting cache size to minimum legal value of %d MB (%d cache entries)\n", cfg_max_cache_count * NNCache::ENTRY_SIZE / 1048576LL, cfg_max_cache_count);
+                    } else {
+                        gtp_printf(id, "Set max cache size to %d MB (%d cache entries)\n", cfg_max_cache_count * NNCache::ENTRY_SIZE / 1048576LL, cfg_max_cache_count);
+                    }
+                    return true;
+                } else {
+                    gtp_fail_printf(id, "incorrect value for cache size");
+                    return true;
+                }
+            }
+        }
+        std::string options_out_tmp("");
+        for (int i = 0; s_options[i].size() > 0; i++) {
+            options_out_tmp = options_out_tmp + "\n" + s_options[i];
+        }
+        gtp_printf(id, options_out_tmp.c_str());
```

Comment:
Is this for debugging?

Reply:
set_option without arguments is printing available options